internal/controlplane/handlers_oauth.go

```diff
@@ -115,55 +127,93 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 	}, nil
 }
 
-// ExchangeCodeForTokenCLI exchanges an OAuth2 code for a token
-// This function gathers the state from the database and compares it to the state
-// passed in. If they match, the code is exchanged for a token.
-// This function is used by the CLI client.
-func (s *Server) ExchangeCodeForTokenCLI(ctx context.Context,
-	in *pb.ExchangeCodeForTokenCLIRequest) (*httpbody.HttpBody, error) {
+// HandleProviderCallback handles the OAuth 2.0 authorization code callback from the enrolled
+// provider. This function gathers the state from the database and compares it to the state
+// passed in. If they match, the provider code is exchanged for a provider token.
+// note: this is an HTTP only (not RPC) handler
+func (s *Server) HandleProviderCallback() runtime.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, pathParams map[string]string) {
+		ctx := context.Background()
```

Comment:
This should use [`r.Context()`](https://pkg.go.dev/net/http#Request.Context), not the background context.  (Or it can just pass along `r`.)